br/pkg/lightning/restore/table_restore.go

```diff
@@ -1021,3 +1026,31 @@ func estimateCompactionThreshold(cp *checkpoints.TableCheckpoint, factor int64)
 
 	return threshold
 }
+
+func (tr *TableRestore) allocateRowIDs(newRowCount int64, rc *Controller) (int64, int64, error) {
+	tr.rowIDLock.Lock()
+	defer tr.rowIDLock.Unlock()
+	metaMgr := rc.metaMgrBuilder.TableMetaMgr(tr)
+	// try to re-allocate from downstream
+	// if we are using parallel import, rowID should be reconciled globally.
+	// Otherwise, this function will simply return 0.
+	preRowIDMax, newRowIDMax, err := metaMgr.ReallocTableRowIDs(context.Background(), newRowCount)
```

Comment:
```suggestion
	newRowIDBase, newRowIDMax, err := metaMgr.ReallocTableRowIDs(context.Background(), newRowCount)
```

Reply:
ReallocTableRowIDs returns the previous maximum rowID though (caller has to +1 by itself). Perhaps I will make it return the new rowID base.